src/App/App.js

```diff
@@ -67,17 +70,18 @@ class App extends Component {
           determineLogButtonStatus={this.determineLogButtonStatus}
         />
         <Route exact path='/'>
-          <MovieContainer allMovies={this.state.movies} />
+          <MovieContainer allMovies={this.state.movies} determineShowPageButton={this.determineShowPageButton}/>
         </Route>
-        {/* <Route
-         path='/login'
-        render={(props) => (
-        <Login {...props} setUser={this.setUser} />
-  )}
-/> */}
+        <Route path='/movies/:id'
+          render={({ match }) => {
+            const { id } = match.params;
+            const singleMovie = this.state.movies.find(movie => movie.id === parseInt(id));
+            return <ShowPage {...singleMovie} />
```

Comment:
Still wondering if returning the `ShowPage` component is part of our problem.

Reply:
Side note: maybe we should try adding `component={ ShowPage }` as part of the self-closing `Route` starting on 75.

Reply:
We'll see in the lesson tomorrow - you need either render or component. In this case, render seems right.